document-loaders/langchain4j-document-loader-azure-storage-blob/src/main/java/dev/langchain4j/data/document/source/azure/storage/blob/AzureBlobStorageSource.java

```diff
@@ -0,0 +1,45 @@
+package dev.langchain4j.data.document.source.azure.storage.blob;
+
+import com.azure.storage.blob.models.BlobProperties;
+import dev.langchain4j.data.document.DocumentSource;
+import dev.langchain4j.data.document.Metadata;
+
+import java.io.InputStream;
+
+import static dev.langchain4j.internal.ValidationUtils.ensureNotBlank;
+import static dev.langchain4j.internal.ValidationUtils.ensureNotNull;
+import static java.lang.String.format;
+
+public class AzureBlobStorageSource implements DocumentSource {
+
+    public static final String SOURCE = "source";
+
+    private final InputStream inputStream;
+    private final String accountName;
+    private final String containerName;
+    private final String blobName;
+    private final BlobProperties properties;
+
+    public AzureBlobStorageSource(InputStream inputStream, String containerName, String accountName, String blobNAme, BlobProperties properties) {
+        this.inputStream = ensureNotNull(inputStream, "inputStream");
+        this.accountName = ensureNotBlank(containerName, "accountName");
+        this.containerName = ensureNotBlank(containerName, "containerName");
+        this.blobName = ensureNotBlank(blobNAme, "blobName");
+        this.properties = ensureNotNull(properties, "properties");
+    }
+
+    @Override
+    public InputStream inputStream() {
+        return inputStream;
+    }
+
+    @Override
+    public Metadata metadata() {
+        Metadata metadata = new Metadata();
+        metadata.add(SOURCE, format("https://%s.blob.core.windows.net/%s/%s", accountName, containerName, blobName));
+        metadata.add("Creation-Time", properties.getCreationTime());
```

Comment:
WDYT about prepending metadata keys with "azure_blob_" suffix and converting into snake_case for uniformity with other loaders?

Reply:
I'm using `azure_storage_blob_`